src/server/api/controllers/vessel.controller.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+const HttpError = require("../lib/utils/http-error");
+const knex = require("../../config/db");
+const bcrypt = require("bcrypt");
+const generatePassword = require("generate-password");
+const mailer = require("../lib/services/mailer.service");
+const { hashPassword } = require("../lib/utils/hash.password");
+//GET
+const getVessels = async req => {
+  try {
+    return await knex("vessels").select(
+      "vessels.id",
+      "vessels.organization_id",
+      "vessels.mmsi",
+      "vessels.name"
+    );
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+//GET:id
+const getVesselById = async id => {
+  try {
+    const vessels = await knex("vessels")
+      .select("vessels.id as id", "organization_id", "mmsi", "name")
+      .where({ id: id });
+    if (vessels.length === 0) {
+      throw new HttpError("Bad request", `Cannot find user for ID ${id}!`, 404);
+    }
+    return vessels;
+  } catch (err) {
+    return err.message;
+  }
+};
+//POST
+
+const createVessel = async ({ body }) => {
+  const { id, organization_id, mmsi, name } = body;
```

Comment:
You can remove `id` from that destructuring statement, since you are newer using it and it cause potential to confuse it with `organization_id`. When you create a new record you don't know the ID. The DB will provide that for you.

Reply:
removed id from create vessel